src/message/mailbox/parsers/rfc2822.rs

```diff
@@ -0,0 +1,353 @@
+    choice((
+        rfc2234::alpha(),
+        rfc2234::digit(),
+        filter(|c| {
+            matches!(
+                *c,
+                '!' | '#'
+                    | '$'
+                    | '%'
+                    | '&'
+                    | '\''
+                    | '*'
+                    | '+'
+                    | '-'
+                    | '/'
+                    | '='
+                    | '?'
+                    | '^'
+                    | '_'
+                    | '`'
+                    | '{'
+                    | '|'
+                    | '}'
+                    | '~'
+            )
+        }),
+        // also allow non ASCII UTF8 chars
+        rfc5336::utf8_non_ascii(),
+    ))
+}
+
+// atom            =       [CFWS] 1*atext [CFWS]
+pub(super) fn atom() -> impl Parser<char, Vec<char>, Error = Simple<char>> {
+    // NOTE: the last CFWS can be skipped since atoms are only used in
+    // serie inside display names, so the last CFWS will always be
+    // captured together with the first CFWS. This prevents to capture
+    // trailing CFWS.
+    cfws()
+        .or_not()
+        .map(Option::unwrap_or_default)
+        .chain(atext().repeated().at_least(1))
+        .chain(cfws().or_not().map(Option::unwrap_or_default))
+}
+
+// dot-atom        =       [CFWS] dot-atom-text [CFWS]
+pub fn dot_atom() -> impl Parser<char, Vec<char>, Error = Simple<char>> {
+    cfws()
+        .or_not()
+        .map(Option::unwrap_or_default)
+        .chain(dot_atom_text())
+        .chain(cfws().or_not().map(Option::unwrap_or_default))
+}
+
+// dot-atom-text   =       1*atext *("." 1*atext)
+pub fn dot_atom_text() -> impl Parser<char, Vec<char>, Error = Simple<char>> {
+    atext().repeated().at_least(1).chain(
+        just('.')
+            .chain(atext().repeated().at_least(1))
+            .repeated()
+            .at_least(1)
+            .flatten(),
+    )
+}
+
+// 3.2.5. Quoted strings
+// https://datatracker.ietf.org/doc/html/rfc2822#section-3.2.5
+
+// qtext           =       NO-WS-CTL /     ; Non white space controls
+//
+//                         %d33 /          ; The rest of the US-ASCII
+//                         %d35-91 /       ;  characters not including "\"
+//                         %d93-126        ;  or the quote character
+fn qtext() -> impl Parser<char, char, Error = Simple<char>> {
+    choice((
+        no_ws_ctl(),
+        filter(|c| matches!(u32::from(*c), 33 | 35..=91 | 93..=126)),
+    ))
+}
+
+// qcontent        =       qtext / quoted-pair
+pub(super) fn qcontent() -> impl Parser<char, char, Error = Simple<char>> {
+    choice((qtext(), quoted_pair(), rfc5336::utf8_non_ascii()))
+}
+
+// quoted-string   =       [CFWS]
+//                         DQUOTE *([FWS] qcontent) [FWS] DQUOTE
+//                         [CFWS]
+fn quoted_string() -> impl Parser<char, Vec<char>, Error = Simple<char>> {
+    cfws()
+        .or_not()
+        .map(Option::unwrap_or_default)
```

Comment:
Is this keeping the surrounding whitespace as part of the string? I think it should be dropped and use `ignore_then` instead of `chain`.

Reply:
Yes indeed, I forgot to remove it after my tests :heavy_check_mark: